lib/cylc/gui/suite_log_viewer.py

```diff
@@ -84,77 +91,61 @@ def create_gui_panel(self):
         self.window.show_all()
 
     def shutdown(self, w, e, wind):
+        """Quite the suite log viewer."""
         self.quit()
         wind.destroy()
 
     def filter_log(self, cb):
+        """Filter for task names."""
         model = cb.get_model()
         index = cb.get_active()
         if index == 0:
             return False
-
         task = model[index][0]
         if task == 'all':
             filter_ = None
         else:
             filter_ = '\\[' + task + '%\d+\\]'
-
         self.task_filter = filter_
         self.update_view()
-
-        # TODO - CHECK ALL BOOLEAN RETURN VALUES THROUGHOUT THE GUI
         return False
 
     def custom_filter_log(self, e):
+        """Filter for arbitrary text."""
         txt = e.get_text()
         if txt == '':
             filter_ = None
         else:
             filter_ = txt
-
         self.custom_filter = filter_
         self.update_view()
-
         return False
 
-    def current_log(self):
-        try:
-            return get_logs(self.dirname, self.main_log, False)[self.level]
-        except IndexError:
-            return None
-
     def rotate_log(self, bt, go_older):
+        """Switch to other log rotations."""
         if go_older:
```

Comment:
We are protecting the lower one - it can't go below zero.  We aren't protecting the upper one, because the GUI doesn't know what the oldest rotation is.  However, the cat-log command now prints an error message if you request a non-existent rotation, and that gets shown in the GUI window.